CMakeLists.txt

```diff
@@ -466,18 +491,18 @@ if(CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR)
 		add_executable(docsparser samples/docs_parser.cpp)
 		add_executable(scratch samples/scratch.cpp)
 		add_executable(https samples/https.c)
-		add_executable(spaceshooter samples/spaceshooter.cpp)
+		add_executable(spaceshooter samples/spaceshooter.cpp samples/spaceshooter_data/flash_shd.h)
 		add_executable(draw_to_texture samples/draw_to_texture.c)
-		add_executable(hello_triangle samples/hello_triangle.c)
-		add_executable(waves samples/waves.cpp)
-		add_executable(shallow_water samples/shallow_water.cpp)
+		add_executable(hello_triangle samples/hello_triangle.c samples/hello_triangle_data/tri_fs.h samples/hello_triangle_data/tri_vs.h)
+		add_executable(waves samples/waves.cpp samples/waves_data/waves_shd.h)
+		add_executable(shallow_water samples/shallow_water.cpp samples/shallow_water_data/shader.h)
 		add_executable(noise samples/noise.c)
 		add_executable(fetch_image samples/fetch_image.cpp)
-		add_executable(metaballs samples/metaballs.cpp)
+		add_executable(metaballs samples/metaballs.cpp samples/metaballs_data/shader_shd.h)
 		add_executable(timestep samples/timestep.cpp)
 		add_executable(joypad samples/joypad.c)
 		add_executable(outline_stencil samples/outline_stencil.cpp)
-		add_executable(recolor samples/recolor.cpp)
+		add_executable(recolor samples/recolor.cpp samples/recolor_data/recolor_shd.h)
```

Comment:
I would prefer the majority of sample code to use runtime string shaders. Precompiling is unneeded complexity when starting out.

Reply:
Oh I made the changes just to check that all samples are fine with `CF_GLSLANG=OFF`

I can revert but those samples would crash when `CF_GLSLANG=OFF`, is that acceptable?

Reply:
Yep that's fine. As long as they all work with default settings when cloning, if the user turns it off I would expect them to then crash.